python/foglamp/services/core/api/service.py

```diff
@@ -236,14 +280,34 @@ def get_service_records():
     return result['count']
 
 
-async def revert_configuration(storage, key):
+async def get_schedule(storage, schedule_name):
+    payload = PayloadBuilder().SELECT(["id", "enabled"]).WHERE(['schedule_name', '=', schedule_name]).payload()
+    result = await storage.query_tbl_with_payload('schedules', payload)
+    return result
+
+
+async def delete_configuration(storage, key):
+    await delete_configuration_category(storage, key)
+    await delete_configuration_category(storage, "{}Advanced".format(key))
```

Comment:
It should not just be a fixed name. What I am saying is that you need to find all the children of this category and delete them all. So go to the category_children table, get all children of this category and delete all of those categories. For each of those categories repeat this until you find no new children. Then remove the entries from the category_children table.

This code should not have any hardwired name, like Advanced in it all all.

Reply:
Well, {}Advanced is being deleted from `configuration` table itself. which is required as we have that hardwired insertion via south->ingest.py `        category = "{}Advanced".format(cls._parent_service._name)`

> get all children of this category and delete all of those categories

next 2 def calls are doing the same: delete_parent_child_configuration, delete_advance_child_configuration

I did not find any config thing (including) left in `configuration` or `category_children` table after these calls.

Reply:
@MarkRiddoch  FOGL-2014 created.